src/context/mod.rs

```diff
@@ -217,7 +256,177 @@ pub trait ContextTimers<'ctx>: private::SealedContext<'ctx> {
+/// **Tip: No need for message queues**:
+/// Without Restate, you would usually put a message queue in between the two services, to guarantee the message delivery.
+/// Restate eliminates the need for a message queue because Restate durably logs the request and makes sure it gets executed.
+///
+/// ## Delayed calls
+///
+/// A delayed call is a one-way call that gets executed after a specified delay.
+///
+/// To schedule a delayed call, send a message with a delay parameter, as follows:
+///
+/// ```rust,no_run
+/// # #[path = "../../examples/services/mod.rs"]
+/// # mod services;
+/// # use services::my_virtual_object::MyVirtualObjectClient;
+/// # use services::my_service::MyServiceClient;
+/// # use services::my_workflow::MyWorkflowClient;
+/// # use restate_sdk::prelude::*;
+/// # use std::time::Duration;
+/// #
+/// # async fn greet(ctx: Context<'_>, greeting: String) -> Result<(), HandlerError> {
+///     // To a Service:
+///     ctx.service_client::<MyServiceClient>()
+///         .my_handler(String::from("Hi!"))
+///         .send_with_delay(Duration::from_millis(5000));
+///
+///     // To a Virtual Object:
+///     ctx.object_client::<MyVirtualObjectClient>("Mary")
+///         .my_handler(String::from("Hi!"))
+///         .send_with_delay(Duration::from_millis(5000));
+///
+///     // To a Workflow:
+///     ctx.workflow_client::<MyWorkflowClient>("my-workflow-id")
+///         .run(String::from("Hi!"))
+///         .send_with_delay(Duration::from_millis(5000));
+///     ctx.workflow_client::<MyWorkflowClient>("my-workflow-id")
+///         .interact_with_workflow()
+///         .send_with_delay(Duration::from_millis(5000));
+///  #    Ok(())
+///  # }
+/// ```
+///
+/// **Tip: Scheduling async tasks**:
+/// You can also use this functionality to schedule async tasks.
+/// Restate will make sure the task gets executed at the desired time.
+///
+/// **Note: Ordering guarantees in Virtual Objects**:
+/// Invocations to a Virtual Object are executed serially.
+/// Invocations will execute in the same order in which they arrive at Restate.
+/// For example, assume a handler calls the same Virtual Object twice:
+///
+/// ```rust,no_run
+/// # #[path = "../../examples/services/my_virtual_object.rs"]
+/// # mod my_virtual_object;
+/// # use my_virtual_object::MyVirtualObjectClient;
+/// # use restate_sdk::prelude::*;
+/// # async fn greet(ctx: Context<'_>, greeting: String) -> Result<(), HandlerError> {
+///     ctx.object_client::<MyVirtualObjectClient>("Mary")
+///         .my_handler(String::from("I'm call A!"))
+///         .send();
+///     ctx.object_client::<MyVirtualObjectClient>("Mary")
+///         .my_handler(String::from("I'm call B!"))
+///         .send();
+/// #    Ok(())
+/// }
+/// ```
+///
+/// It is guaranteed that call A will execute before call B.
+/// It is not guaranteed though that call B will be executed immediately after call A, as invocations coming from other handlers/sources, could interleave these two calls.
```

Comment:
This is again another thing that IMO should be in the general description of virtual objects